nifi-nar-bundles/nifi-ignite-bundle/nifi-ignite-processors/src/main/java/org/apache/nifi/processors/ignite/cache/PutIgniteCache.java

```diff
@@ -0,0 +1,373 @@
+    "The processor uses the value of FlowFile attribute " + PutIgniteCache.IGNITE_CACHE_ENTRY_KEY + " as the " +
+    "cache key and the byte array of the FlowFile as the value of the cache entry value.  Both the string key and a " +
+    " non-empty byte array value are required otherwise the FlowFile is transfered to the failure relation.")
+@WritesAttributes({
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_TOTAL_COUNT, description = "The total number of FlowFile in the batch"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_ITEM_NUMBER, description = "The item number of FlowFile in the batch"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_SUCCESSFUL_ITEM_NUMBER, description = "The successful FlowFile item number"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_SUCCESSFUL_COUNT, description = "The number of successful FlowFiles"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_FAILED_ITEM_NUMBER, description = "The failed FlowFile item number"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_FAILED_COUNT, description = "The total number of failed FlowFiles in the batch"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_FAILED_REASON_ATTRIBUTE_KEY, description = "The failed reason attribute key")
+    })
+@ReadsAttributes({
+    @ReadsAttribute(attribute = PutIgniteCache.IGNITE_CACHE_ENTRY_KEY, description = "Ignite cache key"),
+    })
+public class PutIgniteCache extends AbstractIgniteCacheProcessor {
+
+    /**
+     * The batch size of flow files to be processed on invocation of onTrigger
+     */
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size For Entries")
+            .description("Batch size for entries (1-500).")
+            .defaultValue("250")
+            .required(false)
+            .addValidator(StandardValidators.createLongValidator(1, 500, true))
+            .sensitive(false)
+            .build();
+
+    /**
+     * Data streamer's per node parallelism
+     */
+    public static final PropertyDescriptor DATA_STREAMER_PER_NODE_PARALLEL_OPERATIONS = new PropertyDescriptor.Builder()
+            .name("Data Streamer Per Node Parallel Operations")
+            .description("Data streamer per node parallelism")
+            .defaultValue("5")
+            .required(false)
+            .addValidator(StandardValidators.createLongValidator(1, 10, true))
+            .sensitive(false)
+            .build();
+
+    /**
+     * Data streamers per node buffer size
+     */
+    public static final PropertyDescriptor DATA_STREAMER_PER_NODE_BUFFER_SIZE = new PropertyDescriptor.Builder()
+            .name("Data Streamer Per Node Buffer Size")
+            .description("Data streamer per node buffer size (1-500).")
```

Comment:
The buffer size is the number of objects to buffer.  Let me know what you think.